addons/html_builder/static/src/builder/snippets_menu.js

```diff
@@ -103,14 +112,77 @@ export class SnippetsMenu extends Component {
         });
     }
 
+    /**
+     * Handles the flag of the closest savable element to the mutation as dirty
+     *
+     * @param {Object} records - The observed mutations
+     * @param {Boolean} undoOperation - True if mutations are done due to an
+     * undo operation.
+     */
+    handleMutations(records, undoOperation) {
+        if (undoOperation) {
+            // Do nothing as `o_dirty` has already been handled by the history
+            // plugin
+            return;
+        }
+        for (const record of records) {
+            if (record.attributeName === "contenteditable") {
+                continue;
+            }
+            let targetEl = record.target;
+            if (!targetEl.isConnected) {
+                continue;
+            }
+            if (targetEl.nodeType !== Node.ELEMENT_NODE) {
+                targetEl = targetEl.parentElement;
+            }
+            if (!targetEl) {
+                continue;
+            }
+            const savableEl = targetEl.closest(SAVABLE_SELECTOR);
+            if (
+                !savableEl ||
+                savableEl.classList.contains("o_dirty") ||
+                savableEl.hasAttribute("data-oe-readonly")
+            ) {
+                return;
+            }
+            savableEl.classList.add("o_dirty");
+            break;
+        }
+    }
+
+    get context() {
+        return Object.assign({}, user.context, {
+            website_id: this.websiteService.currentWebsite.id,
+            lang: this.websiteService.currentWebsite.metadata.lang,
+            user_lang: user.context.lang,
+        });
+    }
+
     discard() {
         // TODO: adapt
         this.editor.getContent();
         this.props.closeEditor();
     }
 
-    save() {
-        console.log("todo");
+    async save() {
+        const editableEls = this.getContentEditableAreas(this.editor);
+        const defs = editableEls.map((editableEl) =>
+            this.editor.shared.savePendingImages(editableEl)
+        );
+        await Promise.all(defs);
```

Comment:
await Promise.all(editableEls.map((editableEl) =>
            this.editor.shared.savePendingImages(editableEl)
        ))